includes/wf_me_discount_amount_helper.inc

```diff
@@ -0,0 +1,173 @@
+<?php
+
+
+/**
+ * Class wf_me_discount_amount_helper
+ *
+ * This class helps with calculating actual discounted amount for
+ * a membership type given a discount code. It encapsulates logic
+ * for retrieving discount code details and calculating the price for
+ * each membership depending on if the discount code is fixed or
+ * percentage amount.
+ */
+class wf_me_discount_amount_helper {
+
+  const WEBFORM_DISCOUNT_PERCENTAGE_AMOUNT = 1;
+  const WEBFORM_DISCOUNT_FIXED_AMOUNT = 2;
+
+  /**
+   * @var array
+   */
+  private $formStorage;
+
+  private $numberOfInstallments;
+
+  /**
+   * wf_me_discount_amount_helper constructor.
+   *
+   * @param array $formStorage
+   */
+  public function __construct($formStorage) {
+    $this->formStorage = $formStorage;
+  }
+
+  /**
+   * Returns the adjusted webform line Items after re-calculating the amount for
+   * the membership based on the discount code. Returns NULL if there are no matching
+   * membership types when comparing the user membership types and the membership types
+   * linked to the discount code or if the discount code has no membership types linked.
+   *
+   * @param string $discountCode
+   * @param array $lineItems
+   * @param array $membershipTypesAdded
+   *
+   * @return array|NULL
+   */
+  public function getAdjustedLineItems($discountCode, $lineItems, $membershipTypesAdded) {
+    $civiDiscountExtensionHelper = new wf_me_cividiscount_ext_helper();
+    $discountCodeDetails = $civiDiscountExtensionHelper->getDiscountCodeDetails($discountCode);
+
+    if (empty($discountCodeDetails['memberships'])) {
```

Comment:
Kind of a long method here. Might I suggest to clean it up by breaking it into smaller chunks?

Eg:
```php
public function getAdjustedLineItems($discountCode, $lineItems, $membershipTypesAdded) {
  $discountCodeDetails = $this->getDiscountDetails($discountCode);
  if (empty($discountCodeDetails['memberships'])) {
    return NULL;
  }

  $membershipNamesForDiscount = $this->getMembershipNamesForDiscount($discountCodeDetails['memberships']);
  if (!array_intersect(array_keys($membershipNamesForDiscount), $membershipTypesAdded)) {
    return NULL;
  }

  return $this->adjustLineItems($lineItems, $membershipNamesForDiscount);
}
```